pool/example_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package pool_test
+
+import "github.com/m3db/m3x/pool"
+
+type exampleObject struct {
+	a, b, c int64
+}
+
+func (o *exampleObject) reset() {
+	o.a = 0
+	o.b = 0
+	o.c = 0
+}
+
+func ExampleObjectPool() {
```

Comment:
same as above, need an `// Output` block

Reply:
This certainly should be run. Will update.

Reply:
Updated this as well, GIthub didn't notice because this particular line didn't change.

Reply:
nit: could you drive the output off the retrieved object, something like -

```
fmt.Printf("Retrieved struct should have default values: %+v", o) 
// Output: ... 
```